lib.php

```diff
@@ -359,33 +453,104 @@ function zoom_calendar_item_update(stdClass $zoom) {
     global $CFG, $DB;
     require_once($CFG->dirroot.'/calendar/lib.php');
 
+    if (!$zoom->recurring) {
+        $eventid = $DB->get_field('event', 'id', array(
+            'modulename' => 'zoom',
+            'instance' => $zoom->id,
+        ));
+    
+        // Load existing event object, or create a new one.
+        $event = zoom_populate_calender_item($zoom);
+        if (!empty($eventid)) {
+            calendar_event::load($eventid)->update($event);
+        } else {
+            calendar_event::create($event);
+        }
+    } else {
+        // First remove existing events for recurring meeting.
+        zoom_calendar_item_delete($zoom);
+        // Based on data passed back from zoom, create the calendar events.
+        if (!empty($zoom->occurrences)) {
+            foreach ($zoom->occurrences as $occurence) {
+                $event = zoom_populate_calender_item($zoom, $occurence);
+                calendar_event::create($event);
+            }
+        }
+    }
+}
+
+/**
+ * Return an array with the days of the week.
+ * 
+ * @return array
+ */
+function zoom_get_weekday_options() {
+    return [
+        1 => get_string('sunday', 'calendar'),
+        2 => get_string('monday', 'calendar'),
+        3 => get_string('tuesday', 'calendar'),
+        4 => get_string('wednesday', 'calendar'),
+        5 => get_string('thursday', 'calendar'),
+        6 => get_string('friday', 'calendar'),
+        7 => get_string('saturday', 'calendar'),
+    ];
+}
+
+/**
+ * Return an array with the weeks of the month.
+ * 
+ * @return array
+ */
+function zoom_get_monthweek_options() {
+    return [
+        1 => get_string('weekoption_first', 'zoom'),
+        2 => get_string('weekoption_second', 'zoom'),
+        3 => get_string('weekoption_third', 'zoom'),
+        4 => get_string('weekoption_fourth', 'zoom'),
+        -1 => get_string('weekoption_last', 'zoom'),
+    ];
+}
+
+/**
+ * Populate the calendar event object, based on the zoom instance
+ * 
+ * @param stdClass $zoom The zoom instance.
+ * @param stdClass $occurrence The occurrence object passed from the zoom api.
+ * @return stdClass The calendar event object.
+ */
+function zoom_populate_calender_item(stdClass $zoom, stdClass $occurrence = null) {
     $event = new stdClass();
     $event->type = CALENDAR_EVENT_TYPE_ACTION;
-    $event->timesort = $zoom->start_time;
+    $event->modulename = 'zoom';
+    $event->eventtype = 'zoom';
+    $event->courseid = $zoom->course;
+    $event->instance = $zoom->id;
+    $event->visible = true;
+    if (!$occurrence) {
+        $event->timesort = $zoom->start_time;
+    } else {
+        $event->timesort = strtotime($occurrence->start_time);
```

Comment:
I think you already perform the strtotime() in populate_zoom_from_response()
```suggestion
        $event->timesort = $occurrence->start_time;
```

Reply:
Thanks, I have removed the unneccesary `strtotime`